src/utils.rs

```diff
@@ -10,177 +23,154 @@ pub fn has_attribute(needle: &str, attributes: &[syn::Attribute]) -> bool {
     })
 }
 
+fn check_expression_is_path(expr: Expr) -> Option<ExprPath> {
+    match expr {
+        Expr::Path(expr_path) if !expr_path.path.segments.is_empty() => Some(expr_path),
+        _ => None,
+    }
+}
+
+fn check_token(token: proc_macro2::TokenTree, arg: &str) -> Option<String> {
+    // this detects the '(...)' part in #[serde(rename_all = "UPPERCASE", tag = "type")]
+    // we can use this to get the value of a particular argument
+    // or to see if it exists at all
+
+    // make sure the delimiter is what we're expecting
+    let proc_macro2::TokenTree::Group(group) = token
+    else
+    {
+        return None;
+    };
+
+    // this detects the '(...)' part in #[serde(rename_all = "UPPERCASE", tag = "type")]
+    // we can use this to get the value of a particular argument
+    // or to see if it exists at all
+
+    // make sure the delimiter is what we're expecting
```

Comment:
this comment seems to have been duplicated

Reply:
Should be fixed in the latest commits.